src/liboslcomp/ast.h

```diff
@@ -748,6 +752,164 @@ class ASTreturn_statement final : public ASTNode {
 };
 
 
+class ASTfunction_call final : public ASTNode {
```

Comment:
Why did you move this ?

Reply:
if I remember correctly so I could add a constructor to ASTtype_constructor that takes a ASTfunction_call (for cases when the lexer interprets a constructor as a function call, which happens quite frequently). To do so I had to move the definition of ASTfunction_call before ASTtype_constructor. There might be a way around that, but forward declaration did not work, and I'm just not really all that familiar with the whole C++ name resolution trickery.